Mathlib/Combinatorics/SimpleGraph/ConcreteColorings.lean

```diff
@@ -48,4 +48,66 @@ theorem chromaticNumber_pathGraph (n : ℕ) (h : 2 ≤ n) :
   · simpa only [pathGraph_two_eq_top, chromaticNumber_top] using
       hc.chromaticNumber_mono_of_embedding (pathGraph_two_embedding n h)
 
+/-- In a bicolored graph colors alternate on every path -/
+theorem pathGraph_Hom_coloring {α} (G : SimpleGraph α) (c : G.Coloring Prop) {n : ℕ} (hn : 1 ≤ n)
+    (hom : pathGraph n →g G) (hc0 : c (hom ⟨0, hn⟩)) (u : Fin n) :
+    c (hom u) ↔ Even u.val := by
+  induction n with
+  | zero => exact (Nat.not_succ_le_zero 0 hn).elim
+  | succ n ih =>
+    have hn_cases : n = 0 ∨ 1 ≤ n := or_iff_not_imp_right.mpr Nat.eq_zero_of_not_pos
+    apply Or.elim hn_cases
+    · intro (hn' : n = 0)
+      have hu : u = 0 :=
+        Fin.le_zero_iff.mp (le_of_le_of_eq (Fin.le_val_last u) (congrArg Nat.cast hn'))
+      simp [hu]
+      exact hc0
+    · intro (hn' : 1 ≤ n)
+      let new_hom : pathGraph n →g G :=
+        Hom.comp hom (Hom.pathGraph (Nat.le_add_right n 1))
+      have hhom0 : c (new_hom ⟨0, hn'⟩) := by
+        simp [Hom.pathGraph_val, Hom.pathGraph]
+        exact hc0
+      have h_new_hom := ih hn' new_hom hhom0
+      have hu : u.val < n ∨ u.val = n := le_iff_lt_or_eq.mp u.is_le
+      apply Or.elim hu
+      · intro (hun : u.val < n)
+        exact h_new_hom ⟨u.val, hun⟩
+      · intro (hun : u.val = n)
+        -- c (hom u) ↔ Even ↑u
+        let last : Fin (n + 1) := ⟨n, Nat.lt.base n⟩
+        let prev_last : Fin (n + 1) := ⟨n - 1, Nat.sub_lt_succ n 1⟩
+        have hpgadj : (pathGraph (n + 1)).Adj prev_last last := by
+          simp [pathGraph_adj]
+          exact Or.intro_left (n + 1 = n - 1) (Nat.sub_add_cancel hn')
+        have hGadj : G.Adj (hom prev_last) (hom last) := hom.map_rel hpgadj
+        have h_c_prev_last : c (hom prev_last) ↔ Even (n - 1) :=
+          h_new_hom ⟨n-1, Nat.sub_lt hn' Nat.one_pos⟩
+        have h_c_last : c (hom last) ↔ ¬Even (n - 1) := by
+          have h := eq_iff_iff.not.mp (c.valid hGadj).symm
+          rw [h_c_prev_last] at h
+          exact (not_iff_comm.mp (not_iff.mp h)).symm
+        simp [Fin.eq_mk_iff_val_eq.mpr hun, h_c_last]
+        rw [← @Nat.sub_add_cancel n 1 hn']
+        exact Nat.even_add_one.symm
+
+theorem pathGraph_Hom_coloring' {α} (G : SimpleGraph α) (c : G.Coloring Prop) {n : ℕ} (hn : 1 ≤ n)
+    (hom : pathGraph n →g G) (hc0 : ¬c (hom ⟨0, hn⟩)) (u : Fin n) :
+    c (hom u) ↔ Odd u.val := by
+  let c' : G.Coloring Prop := Coloring.mk (fun v ↦ ¬c v) <| by
+    intro v w
+    intro (h : G.Adj v w)
+    simp
+    rw [not_iff_not]
```

Comment:
This should be a standalone definition:
```
def Coloring.map (f : α ↪ β) : G.Coloring α → G.Coloring β
```
which should itself come from
```
def Embedding.top (f : α ↪ β) : (⊤ : SimpleGraph α) ↪g (⊤ : SimpleGraph β)
```

Reply:
In `Coloring.lean` there is a very similar one: `recolorOfEmbedding {α β : Type*} (f : α ↪ β) : G.Coloring α ↪ G.Coloring`, I will try to use it